specs/features/main_spec.lua

```diff
@@ -0,0 +1,50 @@
+local mock = require('luassert.mock')
+
+local mock_client = {
+	id = 1,
+	name = 'lsp-client-test',
+	request = function(method, params, handler, bufnr)
+	end,
+	request_sync = function(method, params, timeout_ms, bufnr)
+	end,
+	supports_method = function(method)
+	end,
+	setup = function() end
+}
+
+local mock_lsp_config = {
+	log_level = vim.lsp.protocol.MessageType.Warning,
+	message_level = vim.lsp.protocol.MessageType.Warning,
+	settings = vim.empty_dict(),
+	init_options = vim.empty_dict(),
+	handlers = {},
+	autostart = true,
+	capabilities = vim.lsp.protocol.make_client_capabilities(),
+}
+
+vim.lsp.buf_get_clients = function()
+	local clients = {}
+	clients[mock_client.name] = mock_client
+	return clients
+end
+
+local function dump(...)
+	io.stdout:write(tostring(vim.inspect({ ... })))
+end
+
+describe("lsp-format", function()
+	before_each(function()
+		require 'lsp-format'.setup {}
+		require 'lspconfig'.denols.setup {
+			on_attach = function() return mock_client end
+		}
+	end)
+
+	it("formats a file", function()
+		local c = mock(mock_client, true)
+		local bufnr = vim.api.nvim_create_buf(false, true)
+		vim.lsp.buf.format({ async = true })
```

Comment:
That is the built-in format request. lsp-format does not use that one.
And you also need to attach the client.

```suggestion
		require("lsp-format").on_attach(c)
		require("lsp-format").format {}
```